meson.build

```diff
@@ -142,6 +140,46 @@ if git.found()
 endif
 add_project_arguments('-DSWAY_VERSION=@0@'.format(version), language: 'c')
 
+# Compute the relative path used by compiler invocations.
+source_root = meson.source_root().split('/')
+build_root = meson.build_root().split('/')
+relative_dir_parts = []
+i = 0
+in_prefix = true
+foreach p : build_root
+	if i >= source_root.length() or not in_prefix or p != source_root[i]
+		in_prefix = false
+		relative_dir_parts += '..'
+	endif
+	i += 1
+endforeach
+i = 0
+in_prefix = true
+foreach p : source_root
+	if i >= build_root.length() or not in_prefix or build_root[i] != p
+		in_prefix = false
+		relative_dir_parts += p
+	endif
+	i += 1
+endforeach
+relative_dir = join_paths(relative_dir_parts) + '/'
+
+# Strip relative path prefixes from the code if possible, otherwise hide them.
+if cc.has_argument('-fmacro-prefix-map=/prefix/to/hide=')
+	add_project_arguments(
+		'-fmacro-prefix-map=@0@='.format(relative_dir),
+		language: 'c',
+	)
+else
```

Comment:
I'd rather just see this fallback removed.
If a compiler that doesn't support this is being used, they'll just get whatever the path is, which 99% of the time would just be `../src/file.c` instead of `src/file.c`, unless their build directory is outside of the source directory.

That currently includes clang, but there us a patch implementing it. I'm not really sure what's holding it up.

Reply:
I really like having uncluttered logs, especially when testing changes from a narrow terminal; the fallback saves 8 characters per line at minimum with out of tree builds, and possibly more when sway is built with a packaging tool.

Reply:
I'm okay with this but we need to ask meson upstream for a string.length() rather than this hack.

Reply:
It turns out that sizeof("string") gives you the string length at compile time. No length computation in meson is needed at all.